src/components/TradeWidget/index.tsx

```diff
@@ -26,20 +30,50 @@ const SubmitButton = styled.button`
 
 const TradeWidget: React.FC = () => {
   const { networkId } = useWalletConnection()
-  const DAI = _getTokenDetails('DAI', networkId)
-  const USDC = _getTokenDetails('USDC', networkId)
+  const fallBackNetworkId = networkId ? networkId : Network.Mainnet // fallback to mainnet
+  const tokens = useMemo(() => tokenListApi.getTokens(fallBackNetworkId), [fallBackNetworkId])
```

Comment:
I'm curious - wouldn't it be better to use `networkId` from the `useWalletConnection` effect instead? Because the `const` `fallBackNetworkId` won't change if `networkId` changes thus meaning `tokens` will never change

wdyt?

Reply:
That's a very good point! I have not realized it won't change. 

Complementing what I said in the previous comment, my goal for having a fallback is to always have the trade widget fully rendered, even if the user haven't connected a wallet.
When no wallet is connected, the `networkId` is null, and thus no tokens will be rendered.

Currently in our mock implementation it doesn't matter because regardless of the `networkId`, we'll always return the same tokens, but that won't be the case for the non-mock scenario.

Anyway, thanks for catching that, I'll update it.

Reply:
Also, don't memoize primitive data types, it's useless

Reply:
Reverted, since this is a primitive data type.